pallets/msa/src/lib.rs

```diff
@@ -414,7 +414,9 @@ pub mod pallet {
 		/// * [`Error::ProofHasExpired`] - `add_provider_payload` expiration is in the past
 		/// * [`Error::SignatureAlreadySubmitted`] - signature has already been used
 		///
-		#[pallet::weight(T::WeightInfo::create_sponsored_account_with_delegation())]
+		#[pallet::weight(T::WeightInfo::create_sponsored_account_with_delegation(
+			T::MaxSchemaGrantsPerDelegation::get()
```

Comment:
since you are passing max here, you should also return the actual amount in extrinsic return to have accurate numbers